### PR TITLE
ArrayFilterFunctionReturnTypeReturnTypeExtension handle error types

### PR DESCRIPTION
**Pull request by mglaman**
Fixes https://github.com/phpstan/phpstan/issues/10189

### CODE REVIEW

File: src/Type/Php/ArrayFilterFunctionReturnTypeReturnTypeExtension.php [hunk 1]
```diff
@@ -54,9 +55,13 @@ public function getTypeFromFunctionCall(FunctionReflection $functionReflection,
 		}
 
 		$arrayArgType = $scope->getType($arrayArg);
+		$arrayArgType->isArray();
```
Comment (ondrejmirtes) [inline src/Type/Php/ArrayFilterFunctionReturnTypeReturnTypeExtension.php]: This is a no-op.

Comment (ondrejmirtes) [reply src/Type/Php/ArrayFilterFunctionReturnTypeReturnTypeExtension.php]: The right solution is:

```php
$arrays = TypeUtils::toBenevolentUnion($arrayArgType)->getArrays();
$arrayArgType = TypeCombinator::union(...$arrays);
```

**Review by ondrejmirtes — CHANGES_REQUESTED** [review]

File: src/Type/Php/ArrayFilterFunctionReturnTypeReturnTypeExtension.php (hunk 1 above)
Comment (mglaman) [reply src/Type/Php/ArrayFilterFunctionReturnTypeReturnTypeExtension.php]: Ah, whoops that was a debugging line I thought I had deleted. Thanks for the direction, I will try it out.

Comment (ondrejmirtes) [reply src/Type/Php/ArrayFilterFunctionReturnTypeReturnTypeExtension.php]: Or maybe just would work too:

```php
$arrayArgType = TypeUtils::toBenevolentUnion($arrayArgType);
```

Comment (mglaman) [reply src/Type/Php/ArrayFilterFunctionReturnTypeReturnTypeExtension.php]: That worked!

Comment (ondrejmirtes) [reply src/Type/Php/ArrayFilterFunctionReturnTypeReturnTypeExtension.php]: And please listen to the bot 😊 https://github.com/phpstan/phpstan-src/pull/2794#issuecomment-1830755596

Comment (mglaman) [reply src/Type/Php/ArrayFilterFunctionReturnTypeReturnTypeExtension.php]: I'll rebase off of 1.10.x and change the PR, I was waiting for "this is a horrible wrong approach" or not. Rebasing now

Comment (mglaman) [reply src/Type/Php/ArrayFilterFunctionReturnTypeReturnTypeExtension.php]: Okay, rebased off of 1.10.x and changed the PR base

**Comment by ondrejmirtes** [comment]
Thank you!

**Comment by mglaman** [comment]
Thanks for the guidance along the way!

**Comment by mvorisek** [comment]
Fixed also https://github.com/phpstan/phpstan/issues/8638, thank you!

**Pull request merged**